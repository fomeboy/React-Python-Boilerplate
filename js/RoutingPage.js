import React from 'react'

const data = [
  {
    'key': 1,
    'value': 'Item 1',
    'year': 1970
  },
  {
    'key': 2,
    'value': 'Item 2',
    'year': 1976
  },
  {
    'key': 3,
    'value': 'Item 3',
    'year': 1990
  },
  {
    'key': 4,
    'value': 'Item 4',
    'year': 2010
  }
]

const RoutingPage = React.createClass({
  render () {
    return (
      <div>
        <h3>{`Route ${this.props.match.path} is Working!`}</h3>
        {data.map((item, i) => {
          return (
            <h3 key={i}>{item.value}</h3>
          )
        })}
      </div>
    )
  }
})

export default RoutingPage
