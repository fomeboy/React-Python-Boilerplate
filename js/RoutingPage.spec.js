/* globals test, expect */
import React from 'react'
import RoutingPage from './RoutingPage.js'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

test('Routing page snapshot test', () => {
  // <Link> context stub passed as prop
  const component = renderer.create(<RoutingPage match={{path: '/testRoute'}}/>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Routing page enzyme test', () => {
  const component = shallow(<RoutingPage match={{path: '/testRoute'}}/>)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('Test number of rendered components', () => {
  const component = shallow(<RoutingPage match={{path: 'testRoute'}}/>)
  expect(5).toEqual(component.find('h3').length)
})
