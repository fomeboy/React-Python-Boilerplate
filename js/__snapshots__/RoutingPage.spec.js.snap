exports[`test Routing page enzyme test 1`] = `
<div>
  <h3>
    Route /testRoute is Working!
  </h3>
  <h3>
    Item 1
  </h3>
  <h3>
    Item 2
  </h3>
  <h3>
    Item 3
  </h3>
  <h3>
    Item 4
  </h3>
</div>
`;

exports[`test Routing page snapshot test 1`] = `
<div>
  <h3>
    Route /testRoute is Working!
  </h3>
  <h3>
    Item 1
  </h3>
  <h3>
    Item 2
  </h3>
  <h3>
    Item 3
  </h3>
  <h3>
    Item 4
  </h3>
</div>
`;
